src/main/java/com/marklogic/kafka/connect/source/MarkLogicSourceConnector.java

```diff
@@ -28,9 +31,24 @@ public ConfigDef config() {
     @Override
     public void start(final Map<String, String> config) {
         logger.info("Starting MarkLogicSourceConnector");
+        Boolean configuredForDsl = checkForNonEmptyString(config.get(MarkLogicSourceConfig.DSL_QUERY));
+        Boolean configuredForSerialized = checkForNonEmptyString(config.get(MarkLogicSourceConfig.SERIALIZED_QUERY));
+        if ((!(configuredForDsl || configuredForSerialized)) || (configuredForDsl && configuredForSerialized)) {
+            throw new ConfigException(
+                format("Either a DSL Optic query (%s) or a serialized Optic query (%s), but not not both, are required",
+                    MarkLogicSourceConfig.DSL_QUERY, MarkLogicSourceConfig.SERIALIZED_QUERY)
+            );
+        }
         this.config = config;
     }
 
+     static Boolean checkForNonEmptyString(String value) {
```

Comment:
We have a StringUtils on the classpath already - it's the one from Spring via the ml-javaclient-util dependency - so can just use `StringUtils.hasText` instead of creating this function (it's fine if we depend on StringUtils.hasText in many places).

Reply:
darn it, I knew that existed, but I couldn't think of what library it was in and I couldn't find it. Thanks.